ssl/record/rec_layer_s3.c

```diff
@@ -602,6 +612,15 @@ int ssl3_write_bytes(SSL *s, int type, const void *buf_, size_t len,
             }
         }
 
+#ifndef OPENSSL_NO_TLSEXT
+        if (USE_MAX_FRAGMENT_LENGTH_EXT(s->session)) {
+            unsigned int max_fragment_len = GET_MAX_FRAGMENT_LENGTH(s->session);
+            for (j = 0; j < numpipes; j++) {
+                if (pipelens[j] > max_fragment_len)
+                    pipelens[j] = max_fragment_len;
+            }
+        }
+#endif
```

Comment:
This doesn't make any sense? Why don't we just work out the pipeline lengths correctly in the first place, rather than calculate them all based on `s->max_send_fragment`, only to then throw them all away when we realised that `s->max_send_fragment` is too big! Also, what do we do if `split_send_fragment > max_fragment_len`?